### PR TITLE
Allow usage of only the name in get()

### PR DESCRIPTION
**Pull request by TonyRice**

### CODE REVIEW

File: asyncy/hub/sdk/AsyncyHub.py [hunk 1]
```diff
@@ -98,6 +98,10 @@ def get(self, alias=None, owner=None, name=None) -> Service:
 
     def _get(self, alias: str = None, owner: str = None, name: str = None):
         try:
+            if '/' in alias and alias.count("/") == 1:
```
Comment (wilzbach) [inline asyncy/hub/sdk/AsyncyHub.py]:
```suggestion
            if alias is not None and alias.count("/") == 1:
```

No need to be redundant.

Comment (TonyRice) [reply asyncy/hub/sdk/AsyncyHub.py]: on it!

File: tests/asyncy/hub/sdk/AsyncyHub_test.py [hunk 2]
```diff
@@ -110,3 +110,14 @@ def test_caching(mocker):
     actual_service = hub.get(alias='second_service')
 
     assert actual_service is not None
+
+
+def test_get_with_name():
+    hub = AsyncyHub(db_path=tempfile.mkdtemp())
```
Comment (wilzbach) [inline tests/asyncy/hub/sdk/AsyncyHub_test.py]: This is an integration test which will use the actual live hub data. For the `get` method we just want to test the single method and patch everything else out. So sth. along the lines of:

```py
patch.object(Service, 'select')

Service.select().where.assert_called_with(...)
```

Comment (TonyRice) [reply tests/asyncy/hub/sdk/AsyncyHub_test.py]: got it! I'll update shortly

Comment (wilzbach) [reply tests/asyncy/hub/sdk/AsyncyHub_test.py]: BTW we typically split-up the tests folder in `unittests` and `integration` to help with the distinction.
In `unittests` one is required to mock out all calls of the method under test.
Examples:

Unittests:
- https://github.com/storyscript/sls/blob/master/tests/unittests/document.py
- https://github.com/storyscript/storyscript/blob/master/tests/unittests/Bundle.py

Integration:
- https://github.com/storyscript/sls/blob/master/tests/integration/completion/basic.py#L59
- https://github.com/storyscript/storyscript/blob/master/tests/integration/compiler/Functions.py

File: asyncy/hub/sdk/AsyncyHub.py [hunk 3]
```diff
@@ -84,20 +84,24 @@ def get(self, alias=None, owner=None, name=None) -> Service:
                     service = self._get(alias, owner, name)
 
         if service is not None:
-            assert isinstance(service, Service)
             # We store JSON as plain text because some Python installations
             # do not have support for the json1 extension.
             # First encountered on CircleCI.
-            if service.topics is not None:
-                service.topics = json.loads(service.topics)
+            if isinstance(service, Service):
```
Comment (wilzbach) [inline asyncy/hub/sdk/AsyncyHub.py]: We should keep the `assert` here.

File: asyncy/hub/sdk/AsyncyHub.py [hunk 4]
```diff
@@ -84,20 +84,24 @@ def get(self, alias=None, owner=None, name=None) -> Service:
...
-                service.configuration = json.loads(service.configuration)
+                if service.configuration is not None:
+                    service.configuration = json.loads(service.configuration)
 
         return service
 
     def _get(self, alias: str = None, owner: str = None, name: str = None):
         try:
+            if alias is not None and alias.count("/") == 1:
+                owner, name = alias.split('/')
```
Comment (wilzbach) [inline asyncy/hub/sdk/AsyncyHub.py]: Single, doube quote mix here too.

File: tests/asyncy/hub/sdk/AsyncyHub_test.py [hunk 5]
```diff
@@ -110,3 +110,13 @@ def test_caching(mocker):
     actual_service = hub.get(alias='second_service')
 
     assert actual_service is not None
+
+
+def test_get_with_name(mocker):
+
+    hub = AsyncyHub(db_path=tempfile.mkdtemp())
```
Comment (wilzbach) [inline tests/asyncy/hub/sdk/AsyncyHub_test.py]: You can even patch that out too:

```py
mock.patch.object(AsyncyHub, '__init__', return_value=None)
```

**Comment by adnrs96** [comment]
Seems like this PR is no longer relevant. Code seems to be already in the repo though I don't know how. If true can we please close this :)
Thanks!

**Pull request closed without merging**